apps/aefate/src/aefa_fate.erl

```diff
@@ -341,28 +342,35 @@ set_remote_function(Caller, ?FATE_CONTRACT(Pubkey), Function, CheckPayable, Chec
                     ES1 = aefa_engine_state:set_code_cache(CodeCache1, ES),
                     ES2 = aefa_engine_state:set_chain_api(APIState1, ES1),
                     ES3 = aefa_engine_state:update_for_remote_call(Pubkey, ContractCode, VMV, Caller, ES2),
-                    check_flags_and_set_local_function(CheckPayable, CheckPrivate, Function, ES3);
+                    check_flags_and_set_local_function(CheckPayable, CheckPrivate, Function, AllowInit, ES3);
                 error ->
                     abort({trying_to_call_contract, Pubkey}, ES)
             end;
         {ContractCode, VMV} ->
             ES1 = aefa_engine_state:update_for_remote_call(Pubkey, ContractCode, VMV, Caller, ES),
-            check_flags_and_set_local_function(CheckPayable, CheckPrivate, Function, ES1)
+            check_flags_and_set_local_function(CheckPayable, CheckPrivate, Function, AllowInit, ES1)
     end.
 
-check_flags_and_set_local_function(false, false, Function, ES) ->
-    set_local_function(Function, ES);
-check_flags_and_set_local_function(CheckPayable, CheckPrivate, Function, ES) ->
+check_flags_and_set_local_function(false, false, Function, AllowInit, ES) ->
+    set_local_function(Function, AllowInit, ES);
+check_flags_and_set_local_function(CheckPayable, CheckPrivate, Function, AllowInit, ES) ->
     case (not CheckPayable) orelse is_function_payable(Function, ES) of
         true  ->
             case CheckPrivate andalso is_function_private(Function, ES) of
-                false -> set_local_function(Function, ES);
+                false -> set_local_function(Function, AllowInit, ES);
                 true  -> abort({function_is_private, Function}, ES)
             end;
         false -> abort({function_is_not_payable, Function}, ES)
     end.
 
-set_local_function(Function, ES) ->
+set_local_function(Function, true, ES) ->
+    set_local_function_(Function, ES);
+set_local_function(?FATE_INIT_ID, false, ES) ->
+    abort({trying_to_call_function, ?FATE_INIT_ID}, ES);
```

Comment:
Is this error message under consensus or just a local artifact?

Reply:
In either case I would suggest running a "sync from genesis test" before we release IRIS

Reply:
No idea if it is under consensus. Would need to look into how the errors are handled in call tree or some tx tree. I kept the error as before to avoid headache.

Reply:
After some investigation, I don't believe it is under consensus.